Mathlib/Analysis/Distribution/Distribution.lean

```diff
@@ -0,0 +1,185 @@
+/-
+Copyright (c) 2025 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker
+-/
+module
+
+public import Mathlib.Analysis.Distribution.TestFunction
+public import Mathlib.Analysis.LocallyConvex.StrongTopology
+
+/-!
+# Distributions
+
+Let `E` be a real **finite-dimensional normed space**, `Ω` an open subset of `E`,
+and `F` a real **locally convex topological vector space**.
+
+An **`F`-valued distribution on `Ω`** is a continuous `ℝ`-linear map `T : 𝓓(Ω, ℝ) →L[ℝ] F`,
+defined on the space `𝓓(Ω, ℝ)` of real-valued test functions, and taking values in `F`.
+In particular, if `𝕜` is an `RCLike` field, `𝓓'(Ω, 𝕜)` is the usual notion of real or complex
+distribution on `Ω`.
+
+We denote the space of `F`-valued distributions on `Ω` by `𝓓'(Ω, F)`. Topologically,
+it is defined as `𝓓(Ω, ℝ) →L_c[ℝ] F`, meaning that we endow it with topology of uniform
+convergence on compact subsets of `𝓓(Ω, ℝ)`. In this particular case, this happens to coincide
+with the topology of `𝓓(Ω, ℝ) →L[ℝ] F`, namely that of uniform convergence on bounded subsets.
+See the implementation notes below for more details.
+
+Right now, this file contains very few mathematical statements.
+The theory will be expanded very soon.
+
+## Main Declarations
+
+* `𝓓'^{n}(Ω, F) = Distribution Ω F n` is the space of `F`-valued distributions on `Ω` with
+  order at most `n`. See the implementation notes below for more information about the parameter
+  `n : ℕ∞`; in most cases you want to use the space `𝓓'(Ω, F) = Distribution Ω F ⊤`.
+* `Distribution.mapCLM`: any continuous linear map `A : F →L[ℝ] G` induces a continuous linear
+  map `𝓓'(Ω, F) →L[ℝ] 𝓓'(Ω, G)`. On locally integrable functions, this corresponds to applying `A`
+  pointwise.
+
+## Notation
+
+In the `Distributions` scope, we introduce the following notations:
+* `𝓓'^{n}(Ω, F)`: the space of `F`-valued distributions on the open set `Ω` with order at most
+  `n : ℕ∞`.
+* `𝓓'(Ω, F)`: the space of `F`-valued distributions on the open set `Ω`, i.e `𝓓'^{⊤}(Ω, F)`.
+
+## Implementation Notes
+
+### `abbrev` or `def`
+
+At this point in time, it is not clear wether we should enforce an API barrier between
+`𝓓'(Ω, F)` and `𝓓(Ω, ℝ) →L_c[ℝ] F`. For now, we have made the "default" choice to implement
+`Distribution` as an `abbrev`, which means that we get a lot of instances for free, but also
+that there is no API barrier.
```

Comment:
why `barrier` (which I interpret as a negative term)? Perhaps `but also that we cannot develop a specific API for this type`?